tests/dhcp_relay/test_dhcp_relay.py

```diff
@@ -55,6 +56,78 @@ def check_interface_status(duthost):
     return False
 
 
+def query_counter_result(duthost, query_key):
+    counters_query_string = 'sonic-db-cli COUNTERS_DB hgetall "DHCPV4_COUNTER_TABLE:{key}"'
+    result = json.loads(
+        duthost.shell(counters_query_string.format(key=query_key))['stdout'].replace("\"", "").replace("'", "\"")
+    )
+    return {k: {ik: int(iv) for ik, iv in v.items()} for k, v in result.items()}
+
+
+def query_vlan_interface_counters(duthost, vlan_name, interface_name_list):
+    """Format the counters output for the given VLAN and interface names."""
+    if interface_name_list is None or len(interface_name_list) == 0:
+        return query_counter_result(duthost, vlan_name)
+    total_counters = {}
+    for interface_name in interface_name_list:
+        internal_result = query_counter_result(duthost, vlan_name + ":" + interface_name)
+        for key, value in internal_result.items():
+            total_value = total_counters.setdefault(key, {})
+            for sub_key, sub_value in value.items():
+                total_value[sub_key] = total_value.get(sub_key, 0) + sub_value
+    return total_counters
+
+
+def validate_dhcpcom_counters(dhcp_relay, duthost):
+    """Validate the dhcpcom counters"""
+    downlink_vlan_iface = dhcp_relay['downlink_vlan_iface']['name']
+    uplink_portchannels_interfaces = dhcp_relay['uplink_interfaces']
+    client_iface = dhcp_relay['client_iface']['name']
+    dhcp_server_sum = len(dhcp_relay['downlink_vlan_iface']['dhcp_server_addrs'])
+    portchannels = dhcp_relay['portchannels']
+    uplink_interfaces = []
+    for portchannel_name in uplink_portchannels_interfaces:
+        if portchannel_name in portchannels.keys():
+            uplink_interfaces.extend(portchannels[portchannel_name]['members'])
```

Comment:
Please correct me if my understanding is wrong:
If there is pc, then check counter for pc member (Physical interface)
Else, check counter for pc

If so, can we optimize it to check counter for pc too when there is pc?